plugins/out_s3/s3.c

```diff
@@ -50,13 +50,14 @@ static int s3_put_object(struct flb_s3 *ctx, const char *tag, time_t create_time
 
 static int put_all_chunks(struct flb_s3 *ctx);
 
-static void cb_s3_upload(struct flb_config *ctx, void *data);
+static void cb_s3_upload(struct flb_config *ctx, void *data, time_t *timep);
```

Comment:
If you could change all the timep s to something more descriptive like "first_log_time" that would help greatly to clarify your code and logic

Reply:
Actually, could you rename each to either:
```
chunk_first_log_time
```
or
```
file_first_log_time
```

Right now it's really confusing whether you're talking about the chunk's log timestamp (for a single flush) or the file's log timestamp (for multiple flushes) I'm having trouble following the logic

Reply:
will rename